### PR TITLE
Rebrand to celestia-app and use celestia-core

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR rebrands to celestia-app and uses celestai-core along with our fork of the sdk that also uses celestia-core.

closes: #97

### CODE REVIEW

File: cmd/celestia-appd/cmd/app.go
Comment (adlerjohn) [reply cmd/celestia-appd/cmd/app.go:3]: `celestiapp` ofc

**Review by Wondertan — APPROVED** [review]
Haven't found anything __lazy__ left, so approve 👍🏻 

super nit: the name of the PR has a typo

File: app/abci.go
Comment (Bidon15) [inline app/abci.go:9]: After reading your 1st comment and going though the PR, I'm wondering if this is feasable to rename the path to 
`github.com/celestiaorg/celestia-app/x/celestia-app/types` to follow with what we have for celestia-appd
Example of what we are doing it here `github.com/celestiaorg/celestia-app/cmd/celestia-appd/cmd`

**Review by Bidon15 — CHANGES_REQUESTED** [review]

**Review by tzdybal — DISMISSED** [review]
LGTM

File: app/abci.go
Comment (evan-forbes) [reply app/abci.go:9]: I like that option, but was worried about having nested directories with the same name. Also, celestia-appd is not meant to be imported, but celestaiaapp will have to be imported, so the extra - would automatically force packages that import it to rename that import. We should only ever need to include the PayForMessage stuff in it, so maybe we could just name the module `payment`?

we could also go with John's suggestion and call it `celestiapp`

wdyt?

**Review by adlerjohn — APPROVED** [review]

**Comment by adlerjohn** [comment]
Oh I have write access
stutter

File: app/abci.go
Comment (Bidon15) [reply app/abci.go:9]: I like John's suggestion for the rest. Still for this case, payment is more detailed considering your awesome explanation. `payment` lgtm 🌟

Comment (evan-forbes) [reply app/abci.go]: :+1:  resolved in https://github.com/celestiaorg/lazyledger-app/pull/118/commits/b36a13dd83a71196219dd12718a537f6ee1bfeed

**Comment by evan-forbes** [comment]
Sorry @adlerjohn, I made a change after you last approved... mind approving again?

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**
